packages/nimble-components/src/anchor-tabs/index.ts

```diff
@@ -0,0 +1,282 @@
+     * @public
+     */
+    public activetab: HTMLElement | undefined;
+
+    /**
+     * A reference to the tablist div
+     * @internal
+     */
+    public tablist!: HTMLElement;
+
+    private tabIds: string[] = [];
+
+    /**
+     * @internal
+     */
+    public activeidChanged(_oldValue: string, _newValue: string): void {
+        if (this.$fastController.isConnected) {
+            this.setTabs();
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public tabsChanged(): void {
+        if (this.$fastController.isConnected) {
+            this.tabIds = this.getTabIds();
+            this.setTabs();
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public override connectedCallback(): void {
+        super.connectedCallback();
+
+        this.tabIds = this.getTabIds();
+    }
+
+    private readonly isDisabledElement = (el: Element): el is HTMLElement => {
+        return el.getAttribute('aria-disabled') === 'true';
+    };
+
+    private readonly isFocusableElement = (el: Element): el is HTMLElement => {
+        return !this.isDisabledElement(el);
+    };
+
+    private readonly setTabs = (): void => {
+        const gridHorizontalProperty = 'gridColumn';
+        const gridVerticalProperty = 'gridRow';
+
+        this.activetab = undefined;
+        this.tabs.forEach((tab: HTMLElement, index: number) => {
+            const tabId: string = this.tabIds[index]!;
+            const isActiveTab = this.activeid === tabId;
+            const isTabStop = this.activeid === tabId && this.isFocusableElement(tab);
+            tab.setAttribute('id', tabId);
+            tab.setAttribute('aria-selected', isActiveTab ? 'true' : 'false');
+            tab.removeEventListener('click', this.handleTabClick);
+            tab.addEventListener('click', this.handleTabClick);
+            tab.removeEventListener('keydown', this.handleTabKeyDown);
+            tab.addEventListener('keydown', this.handleTabKeyDown);
+            tab.setAttribute('tabindex', isTabStop ? '0' : '-1');
+            if (isActiveTab) {
+                this.activetab = tab;
+            }
+
+            tab.style[gridVerticalProperty] = '';
+            tab.style[gridHorizontalProperty] = `${index + 1}`;
+        });
+    };
+
+    private getTabIds(): string[] {
+        return this.tabs.map((tab: HTMLElement) => {
+            return tab.getAttribute('id') ?? `tab-${uniqueId()}`;
+        });
+    }
+
+    private readonly handleTabClick = (event: MouseEvent): void => {
+        const selectedTab = event.currentTarget as HTMLElement;
+        if (
+            selectedTab.nodeType === 1
+            && this.isFocusableElement(selectedTab)
+        ) {
+            this.tabs.forEach((tab: HTMLElement) => {
+                tab.setAttribute('tabindex', tab === selectedTab ? '0' : '-1');
+            });
+            this.navigateToTab(
+                this.tabs.findIndex((item: HTMLElement) => item === selectedTab)
+            );
```

Comment:
It turns out that it was not needed for clicks, but I was also turning Space/Enter keypresses into clicks on the host element, and it was needed in that case. Instead, I'm now responding to Space/Enter by sending a click to the internal anchor element (instead of the host tab element). This allows me to get rid of the `navigateToTab` function. Note that on a Space/Enter, I do not need to do the `tabindex` updating that the click handler is doing, because the focused tab isn't changing.